docs/debugging.md

```diff
@@ -0,0 +1,117 @@
+# Debugging your scripts locally
+
+For a complete example, see the [debugging example](../examples/debugging/README.md).
+
+- [Debugging your scripts locally](#debugging-your-scripts-locally)
+  - [Dependencies](#dependencies)
+  - [Run your code locally](#run-your-code-locally)
+    - [Use assets from a deployed Substra platform and local assets](#use-assets-from-a-deployed-substra-platform-and-local-assets)
+    - [Dependency between assets](#dependency-between-assets)
+    - [Debugging with logs and breakpoints](#debugging-with-logs-and-breakpoints)
+      - [Access the logs](#access-the-logs)
+      - [Debug with pdb](#debug-with-pdb)
+
+## Dependencies
+
+The required dependencies are:
+
+- Docker:
+  - Mac: [Docker Desktop](https://www.docker.com/products/docker-desktop)
+  - Ubuntu/Debian: `sudo apt install docker docker-compose`
+- Python 3 (recommended 3.6 or 3.7)
+  - It is recommended to use a virtual environment to install Substra packages (for instance [virtualenv](https://virtualenv.pypa.io/en/latest/))
+- Substra package
+
+If you want to debug your script using assets from the Substra platform, you also need access to the platform the assets are on.
+
+
+## Run your code locally
+
+With this installation, you can write scripts using the [SDK](../references/sdk.md) as you would for a remote execution, then
+set the `debug` parameter to `True` when creating the client.
+
+```python
+client = substra.Client.from_config_file(profile_name="node-1", debug=True)
+```
+
+### Use assets from a deployed Substra platform and local assets
+
+With this, you gain access to the platform in 'read-only' mode and any asset you create is created locally.
+
+This means that any function to get, describe or download an asset works with assets from the platform and local assets. Functions to
+list assets will list the assets from the platform and the local ones.  
+Functions that create a new asset (they start with `add`) only create local assets.  
+To differentiate between a local asset and a remote one, look at their key (or pkhash): for local assets, it starts with `local_`.
+
+Local assets are saved in-memory, so they are deleted at the end of the script.
```

Comment:
We could add that the local assets have the same lifetime as the related Client?